server/service/IndividualServicesService.js

```diff
@@ -290,46 +293,68 @@ exports.listApplications = function (user, originator, xCorrelator, traceIndicat
  * customerJourney String Holds information supporting customer’s journey to which the execution applies
  * returns inline_response_200_2
  **/
-exports.listApprovedApplicationsInGenericRepresentation = function (user, originator, xCorrelator, traceIndicator, customerJourney) {
+exports.listApprovedApplicationsInGenericRepresentation = function (user, originator, xCorrelator, traceIndicator, customerJourney, operationServerName) {
   return new Promise(async function (resolve, reject) {
     let response = {};
     try {
       /****************************************************************************************
        * Preparing consequent-action-list for response body
        ****************************************************************************************/
-      let consequentActionList = [];
-
-      /****************************************************************************************
-       * Preparing response-value-list for response body
-       ****************************************************************************************/
-      let responseValueList = [];
-      let applicationProfileList = await profile.getUuidListAsync(applicationProfile.profileNameEnum.APPLICATION_PROFILE);
-      for (let i = 0; i < applicationProfileList.length; i++) {
-        let uuid = applicationProfileList[i];
-        let approvalStatus = await applicationProfile.getApprovalStatusAsync(uuid);
-        if (approvalStatus == applicationProfile.ApplicationProfilePac.ApplicationProfileConfiguration.approvalStatusEnum.APPROVED) {
-          let applicationName = await applicationProfile.getApplicationNameAsync(uuid);
-          let releaseNumber = await applicationProfile.getApplicationReleaseNumberAsync(uuid);
-          let reponseValue = new responseValue(applicationName, releaseNumber, typeof applicationName);
-          responseValueList.push(reponseValue);
+      let consequentActionList = []
+      let responseValueList = []
+      let applicationData = []
+      let filePath
+      let approvalStatus
+      let applicationName
+      let releaseNumber
+      let reponseValue
+      let getOperation
+      let getDataType
+ 
+      // get response profile uuid
+      let responseProfileUuid =  await profile.getUuidListAsync(applicationProfile.profileNameEnum.RESPONSE_PROFILE);
```

Comment:
can we declare a separate function as getDataType and call it at required place to improve code readability

Reply:
done